packages/esm-patient-programs-app/src/programs/programs-form.workspace.tsx

```diff
@@ -52,34 +58,28 @@ const ProgramsForm: React.FC<ProgramsFormProps> = ({
   const { t } = useTranslation();
   const isTablet = useLayoutType() === 'tablet';
   const session = useSession();
-  const availableLocations = useLocations();
-  const { data: availablePrograms } = useAvailablePrograms();
-  const { data: enrollments, mutateEnrollments } = useEnrollments(patientUuid);
-  const [isSubmittingForm, setIsSubmittingForm] = useState(false);
-
+  
   const programsFormSchema = useMemo(() => createProgramsFormSchema(t), [t]);
 
-  const currentEnrollment = programEnrollmentId && enrollments.filter((e) => e.uuid === programEnrollmentId)[0];
-  const currentProgram = currentEnrollment
-    ? {
-        display: currentEnrollment.program.name,
-        ...currentEnrollment.program,
-      }
-    : null;
+  const currentEnrollment = useMemo(
+    () => programEnrollmentId && enrollments.find((e) => e.uuid === programEnrollmentId),
+    [programEnrollmentId, enrollments]
+  );
 
-  const eligiblePrograms = currentProgram
-    ? [currentProgram]
-    : availablePrograms.filter((program) => {
-        const enrollment = enrollments.find((e) => e.program.uuid === program.uuid);
-        return !enrollment || enrollment.dateCompleted !== null;
-      });
+  const currentProgram = useMemo(
+    () => currentEnrollment ? { display: currentEnrollment.program.name, ...currentEnrollment.program } : null,
+    [currentEnrollment]
+  );
 
-  const getLocationUuid = () => {
-    if (!currentEnrollment?.location.uuid && session?.sessionLocation?.uuid) {
-      return session?.sessionLocation?.uuid;
-    }
-    return currentEnrollment?.location.uuid ?? null;
-  };
+  const eligiblePrograms = useMemo(
+    () => currentProgram
+      ? [currentProgram]
+      : availablePrograms.filter((program) => {
+          const enrollment = enrollments.find((e) => e.program.uuid === program.uuid);
+          return !enrollment || enrollment.dateCompleted !== null;
+        }),
+    [availablePrograms, enrollments, currentProgram]
+  );
```

Comment:
Could you kindly remove all the changes that are not related to O3-3793 and leave only the relevant ones?